src/Microsoft.Docs.LearnValidation/Validators/PathValidator.cs

```diff
@@ -26,54 +26,18 @@ public override bool Validate(Dictionary<string, IValidateModel> fullItemsDict)
             {
                 var itemValid = true;
                 var path = item as PathValidateModel;
-                var result = path.ValidateMetadata();
-                if (!string.IsNullOrEmpty(result))
-                {
-                    itemValid = false;
-                    Logger.Log(LearnErrorLevel.Error, LearnErrorCode.TripleCrown_LearningPath_MetadataError, file: item.SourceRelativePath, result);
-                }
-
-                if (path.Achievement == null)
-                {
-                    Logger.Log(LearnErrorLevel.Error, LearnErrorCode.TripleCrown_LearningPath_NoTrophyBind, file: item.SourceRelativePath);
-                }
-                else if (path.Achievement is string achievementUID)
-                {
-                    if (!fullItemsDict.ContainsKey(achievementUID))
-                    {
-                        itemValid = false;
-                        Logger.Log(LearnErrorLevel.Error, LearnErrorCode.TripleCrown_LearningPath_TrophyNotFound, file: item.SourceRelativePath, achievementUID);
-                    }
-                    else if (!(fullItemsDict[achievementUID] is AchievementValidateModel achievement) || achievement.Type != AchievementType.Trophy)
-                    {
-                        itemValid = false;
-                        Logger.Log(LearnErrorLevel.Error, LearnErrorCode.TripleCrown_LearningPath_NonSupportedAchievementType, file: item.SourceRelativePath, achievementUID);
-                    }
-                }
-
-                var childrenCantFind = path.Modules.Where(m => !fullItemsDict.ContainsKey(m) && !LearnValidationHelper.IsModule(m)).ToList();
 
-                var childrenNotModule = path.Modules.Except(childrenCantFind).Where(m =>
-                {
-                    if (!fullItemsDict.ContainsKey(m))
-                    {
-                        return false;
-                    }
-
-                    fullItemsDict[m].Parent = path;
-                    return !(fullItemsDict[m] is ModuleValidateModel);
-                }).ToList();
-
-                if (childrenCantFind.Any())
+                // when trophy is defined in another path, but that path has error when SDP validating
+                if (path.Achievement is string achievementUID && !fullItemsDict.ContainsKey(achievementUID))
                 {
                     itemValid = false;
-                    Logger.Log(LearnErrorLevel.Error, LearnErrorCode.TripleCrown_LearningPath_ChildrenNotFound, file: item.SourceRelativePath, string.Join(",", childrenCantFind));
                 }
 
-                if (childrenNotModule.Any())
+                // path has child module, but that module has error when SDP validating, except the shared module
+                var childrenCantFind = path.Modules.Where(m => !fullItemsDict.ContainsKey(m) && !_externalXrefsCheck(m, "Module")).ToList();
```

Comment:
If SDP check fails, we should be able to pass a list of failed files to here and propagate errors accordingly. I don't quite follow the `_externalXrefsCheck` here.

cc @yishengjin1413

Reply:
Resolve by offline talking.